### PR TITLE
feat: Backport compose

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It backports the changes on compose.py, along with those changes there are minor changes on docker images, scripts, and some tests.

## Why is it important?

Backport things from master to the 7.x branch is tough due compose.py is totally different, this change will make things easier.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ pipeline {
   parameters {
     string(name: 'ELASTIC_STACK_VERSION', defaultValue: "7.10", description: "Elastic Stack Git branch/tag to use")
     string(name: 'BUILD_OPTS', defaultValue: "", description: "Addicional build options to passing compose.py")
-    booleanParam(name: 'Run_As_Master_Branch', defaultValue: false, description: 'Allow to run any steps on a PR, some steps normally only run on master branch.')
+    booleanParam(name: 'Run_As_Master_Branch', defaultValue: true, description: 'Allow to run any steps on a PR, some steps normally only run on master branch.')
```
Comment (kuisathaverat) [inline .ci/Jenkinsfile:36]: to be changed before merge

File: .ci/integrationTestDownstream.groovy [hunk 2]
```diff
@@ -22,7 +22,7 @@ pipeline {
     DISABLE_BUILD_PARALLEL = "${params.DISABLE_BUILD_PARALLEL}"
   }
   options {
-    timeout(time: 1, unit: 'HOURS')
+    timeout(time: 2, unit: 'HOURS')
```
Comment (kuisathaverat) [inline .ci/integrationTestDownstream.groovy:25]: we test 13 versions of the stack in some cases 1 hour is too tight.

TODO: check if we can rid of some of those stack tests (e.g. test only 7.5+)

Comment (kuisathaverat) [reply .ci/integrationTestDownstream.groovy:25]: we can start removing some 

image

**Comment by kuisathaverat** [comment]
Round one, Ruby failed to build the Docker image

```
Installing sprockets-rails 3.2.2
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    can't add a new key into hash during iteration

Gem files will remain installed in /usr/local/bundle/gems/nio4r-2.5.4 for
inspection.
Results logged to
/usr/local/bundle/extensions/x86_64-linux/3.0.0/nio4r-2.5.4/gem_make.out

An error occurred while installing nio4r (2.5.4), and Bundler cannot continue.
Make sure that `gem install nio4r -v '2.5.4' --source 'https://rubygems.org/'`
succeeds before bundling.

In Gemfile:
  rails was resolved to 5.2.4.4, which depends on
    actioncable was resolved to 5.2.4.4, which depends on
      nio4r
```

**Comment by kuisathaverat** [comment]
round two no failures.

**Comment by v1v** [comment]
You might need to backport -> https://github.com/elastic/apm-integration-testing/pull/1041

**Review by cachedout — APPROVED** [review]
So far as I can tell, this all looks safe to backport.

**Pull request merged**